### PR TITLE
Refactored GRR modules

### PR DESCRIPTION
**Pull request by ramo-j**
Refactoring of grr modules to take advantage of parallel processing. Additionally, added support for multiple flows to `GRRFlowCollector` which previously supported only one flow input.

### CODE REVIEW

File: dftimewolf/lib/containers/containers.py [hunk 1]
```diff
@@ -353,6 +353,22 @@ def __init__(self, hostname: str, platform: str = 'unknown') -> None:
     self.platform = platform
 
 
+class GrrHostFlowPair(interface.AttributeContainer):
```
Comment (tomchop) [inline dftimewolf/lib/containers/containers.py:356]: Given that Flows are always associated with a Host (as they are unique *per host*), I think we can just rename this to `GrrHostFlow` or even `GrrFlow`.

File: dftimewolf/lib/collectors/grr_base.py [hunk 2]
```diff
@@ -107,20 +104,20 @@ def _WrapGRRRequestWithApproval(
...
+          logger.success(approval_url) # type: ignore
           time.sleep(self._CHECK_APPROVAL_INTERVAL_SEC)
           continue
 
         # If no approvers were specified, abort.
         if not self.approvers:
           message = ('GRR needs approval but no approvers specified '
                      '(hint: use --approvers)')
-          self.ModuleError(message, critical=True)
+          raise GrrError(message, critical=True) from exception
```
Comment (tomchop) [inline dftimewolf/lib/collectors/grr_base.py:120]: Why not simply raise a `dfTimewolfError` here? You're not catching any `GrrError`s anyways.

**Review by tomchop — CHANGES_REQUESTED** [review]
Some minor changes, otherwise this looks good to me!

File: dftimewolf/lib/collectors/grr_base.py (hunk 2 above)
Comment (ramo-j) [reply dftimewolf/lib/collectors/grr_base.py:120]: `GrrError` is a subclass of `DFTimewolfError`. Added in `Raises` notations.

Comment (tomchop) [reply dftimewolf/lib/collectors/grr_base.py:122]: I meant, why introduce a new `GrrError` class, when dfTimewolf error seems to be enough for this purpose? I don't see anywhere in the code where you're doing `except GrrError as error` (i.e. catching that specific error type)

Comment (ramo-j) [reply dftimewolf/lib/collectors/grr_base.py]: Good point. Removed `GrrError` in favour of `DFTimewolfError`.

**Pull request merged**